widget/table_test.go

```diff
@@ -189,6 +189,239 @@ func TestTable_Refresh(t *testing.T) {
+			assert.Equal(t, tc.want, table.offset)
+			assert.Equal(t, tc.want, table.scroll.Offset)
+		})
+	}
+}
+
+func TestTable_ScrollToBottom(t *testing.T) {
+	test.NewApp()
+	defer test.NewApp()
+
+	// we will test a 20 row x 5 column table where each cell is 50x50
+	// this table has a separator thickness of 1
+	const (
+		maxRows int     = 20
+		maxCols int     = 5
+		width   float32 = 50
+		height  float32 = 50
+	)
+
+	templ := canvas.NewRectangle(color.Gray16{})
+	templ.SetMinSize(fyne.Size{Width: width, Height: height})
+
+	table := NewTable(
+		func() (int, int) { return maxRows, maxCols },
+		func() fyne.CanvasObject { return templ },
+		func(TableCellID, fyne.CanvasObject) {})
+
+	w := test.NewWindow(table)
+	defer w.Close()
+
+	table.ScrollToBottom()
+
+	// element size of 50, plus separator thickness of 1
+	want := fyne.Position{X: 0, Y: 19 * (50 + 1)}
+	assert.Equal(t, want, table.offset)
+	assert.Equal(t, want, table.scroll.Offset)
+}
+
+func TestTable_ScrollToLeading(t *testing.T) {
+	test.NewApp()
+	defer test.NewApp()
+
+	table := NewTable(
+		func() (int, int) { return 3, 5 },
+		func() fyne.CanvasObject {
+			return NewLabel("placeholder")
+		},
+		func(id TableCellID, c fyne.CanvasObject) {
+			text := fmt.Sprintf("Cell %d, %d", id.Row, id.Col)
+			c.(*Label).SetText(text)
+		})
+
+	w := test.NewWindow(table)
+	defer w.Close()
+
+	table.ScrollTo(TableCellID{Row: 8, Col: 4})
+	prev := table.offset
+	want := fyne.Position{X: 0, Y: prev.Y}
+
+	table.ScrollToLeading()
+	assert.Equal(t, want, table.offset)
+	assert.Equal(t, want, table.scroll.Offset)
+}
+
+func TestTable_ScrollToTop(t *testing.T) {
+	test.NewApp()
+	defer test.NewApp()
+
+	const (
+		maxRows int     = 6
+		maxCols int     = 10
+		width   float32 = 50
+		height  float32 = 50
+	)
+
+	templ := canvas.NewRectangle(color.Gray16{})
+	templ.SetMinSize(fyne.Size{Width: width, Height: height})
+
+	table := NewTable(
+		func() (int, int) { return maxRows, maxCols },
+		func() fyne.CanvasObject { return templ },
+		func(TableCellID, fyne.CanvasObject) {})
+
+	w := test.NewWindow(table)
+	defer w.Close()
+
+	table.scrollTo(TableCellID{12, 3})
+	table.ScrollToTop()
+
+	want := fyne.Position{}
+	assert.Equal(t, want, table.offset)
+	assert.Equal(t, want, table.scroll.Offset)
+}
+
+func TestTable_ScrollToTrailing(t *testing.T) {
+	test.NewApp()
+	defer test.NewApp()
+	th := test.NewTheme()
+	test.ApplyTheme(t, th)
+
+	const (
+		maxRows    int     = 20
+		maxCols    int     = 8
+		cellWidth  float32 = 109 // observed from actual output
+		cellHeight float32 = 27  // observed
+	)
+
+	table := NewTable(
+		func() (int, int) { return maxRows, maxCols },
+		func() fyne.CanvasObject {
+			return NewLabel("placeholder")
+		},
+		func(id TableCellID, c fyne.CanvasObject) {
+			text := fmt.Sprintf("Cell %d, %d", id.Row, id.Col)
+			c.(*Label).SetText(text)
+		})
+
+	w := test.NewWindow(table)
+	defer w.Close()
+
+	var (
+		padding float32 = th.Size(theme.SizeNamePadding)
```

Comment:
Isn't this just `theme.Padding()`?

Reply:
I didn't realise that was a thing :) Now I know